src/DynamoCore/Extensions/ExtensionLoader.cs

```diff
@@ -124,5 +132,46 @@ private void Log(string msg)
         /// An event that is raised when an extension starts loading.
         /// </summary>
         public event Action<IExtension> ExtensionLoading;
+
+        internal List<string> directoriesToVerifyCertificates = new List<string>();
+
+        private static bool CheckExtensionCertificates(ExtensionDefinition viewExtension)
+        {
+            //Verify the node library exists in the package bin directory
+            if (!File.Exists(viewExtension.AssemblyPath))
+            {
+                throw new Exception(String.Format(
+                    "An extension called {0} found at {1} is missing dlls which are defined in the view extension definition.  Ignoring it.",
+                    viewExtension.TypeName, viewExtension.AssemblyPath));
+            }
+
+            //Verify that you can load the node library assembly into a Reflection only context
+            Assembly asm;
+            try
+            {
+                asm = Assembly.ReflectionOnlyLoadFrom(viewExtension.AssemblyPath);
+            }
+            catch
+            {
+                throw new Exception(String.Format(
```

Comment:
do these exceptions end up in the console only or are they more visible to the user?

Reply:
I don't know yet.  I will test and verify